osu.Game/Overlays/SocialOverlay.cs

```diff
@@ -62,57 +57,146 @@ public SocialOverlay()
+
+            //currentQuery.ValueChanged += v =>
+            //{
+            //    queryChangedDebounce?.Cancel();
+
+            //    if (string.IsNullOrEmpty(v))
+            //        Scheduler.AddOnce(updateSearch);
+            //    else
+            //    {
+            //        Users = null;
+            //        queryChangedDebounce = Scheduler.AddDelayed(updateSearch, 500);
+            //    }
+            //};
+
+            currentQuery.BindTo(Filter.Search.Current);
+
+            Filter.Tabs.Current.ValueChanged += sortCriteria => Scheduler.AddOnce(updateSearch);
         }
 
         [BackgroundDependencyLoader]
         private void load(APIAccess api)
         {
-            if (Users == null)
-                reloadUsers(api);
+            this.api = api;
+            api.Register(this);
         }
 
-        private void reloadUsers(APIAccess api)
+        private void recreatePanels(PanelDisplayStyle displayStyle)
         {
-            Users = null;
+            if (panels != null)
+            {
+                panels.FadeOut(200);
+                panels.Expire();
+                panels = null;
+            }
 
-            // no this is not the correct data source, but it's something.
-            var request = new GetUsersRequest();
-            request.Success += res =>
+            if (Users == null)
+                return;
+
+            var newPanels = new FillFlowContainer<UserPanel>
             {
-                Users = res.Select(e => e.User);
-                loading.Hide();
+                RelativeSizeAxes = Axes.X,
+                AutoSizeAxes = Axes.Y,
+                Spacing = new Vector2(10f),
+                Margin = new MarginPadding { Top = 10 },
+                ChildrenEnumerable = Users.Select(u =>
+                {
+                    UserPanel panel = new UserPanel(u)
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre
+                    };
+                    switch (displayStyle)
+                    {
+                        case PanelDisplayStyle.Grid:
+                            panel.Width = 300;
+                            break;
+                        default:
+                            panel.RelativeSizeAxes = Axes.X;
+                            break;
+                    }
+                    panel.Status.BindTo(u.Status);
+                    return panel;
+                })
             };
 
-            api.Queue(request);
+            LoadComponentAsync(newPanels, p =>
+            {
+                if (panels != null)
+                    ScrollFlow.Remove(panels);
+
+                ScrollFlow.Add(panels = newPanels);
+            });
+        }
+
+        private APIRequest getUsersRequest;
+
+        private readonly Bindable<string> currentQuery = new Bindable<string>();
+
+        private ScheduledDelegate queryChangedDebounce;

```

Comment:
probably remove this as it isn't used currently

Reply:
remove, as in commenting or actually (temporary) deleting?